src/util.rs

```diff
@@ -1,20 +1,61 @@
+use std::convert::TryFrom;
 use crate::error::ContractError;
-use cosmwasm_std::{QuerierWrapper, Uint128};
-use provwasm_std::{Marker, MarkerType, ProvenanceQuerier, ProvenanceQuery};
+use cosmwasm_std::{Addr, BankMsg, CosmosMsg, Empty, QuerierWrapper, StdError, StdResult, Uint128};
+use provwasm_std::types::provenance::attribute::v1::{Attribute, AttributeQuerier};
+use provwasm_std::types::provenance::marker::v1::{MarkerAccount, MarkerQuerier};
 use rust_decimal::prelude::Zero;
 use rust_decimal::Decimal;
 use uuid::Uuid;
 
-pub fn is_restricted_marker(querier: &QuerierWrapper<ProvenanceQuery>, denom: String) -> bool {
+pub fn is_restricted_marker(querier: &QuerierWrapper, denom: String) -> bool {
     matches!(
-        ProvenanceQuerier::new(querier).get_marker_by_denom(denom),
-        Ok(Marker {
-            marker_type: MarkerType::Restricted,
+        get_marker(denom.clone(), &MarkerQuerier::new(&querier)),
+        Ok(MarkerAccount {
+            marker_type: 2, // 2 is Restricted
             ..
         })
     )
 }
 
+fn get_marker(id: String, querier: &MarkerQuerier<Empty>) -> StdResult<MarkerAccount> {
+    let response = querier.marker(id)?;
+    if let Some(marker) = response.marker {
+        return if let Ok(account) = MarkerAccount::try_from(marker) {
+            Ok(account)
+        } else {
+            Err(StdError::generic_err("unable to type-cast marker account"))
+        };
+    } else {
+        Err(StdError::generic_err("no marker found for id"))
+    }
+}
+
+// TODO: temporary function, need to find alternative function
+pub fn get_attributes(account: String, querier: &AttributeQuerier<Empty>) -> StdResult<Vec<Attribute>> {
+    let response = querier.attributes(account, None)?;
+    Ok(response.attributes)
+}
+
+// TODO: dummy function, need to find alternative function
+pub fn transfer_marker_coins<S: Into<String>, H: Into<Addr>>(
```

Comment:
we need change to https://github.com/provenance-io/provwasm/blob/ef2dba47eaf216a6c7d33698761b96328a277c4b/packages/provwasm-std/src/types/provenance/marker/v1.rs#L1541 for this one

Reply:
here is the thread: 
https://discord.com/channels/897185242182484029/1115703566737342474/1126176055280009278